qt/scientific_interfaces/Indirect/Simulation/DensityOfStates.cpp

```diff
@@ -21,63 +21,56 @@ Mantid::Kernel::Logger g_log("DensityOfStates");
 } // namespace
 
 namespace MantidQt::CustomInterfaces {
+
+enum class DensityOfStates::InputFormat : int { Unsupported = 0, Phonon, Castep, ForceConstants };
+
 DensityOfStates::DensityOfStates(QWidget *parent) : SimulationTab(parent) {
   m_uiForm.setupUi(parent);
+  m_runPresenter = std::make_unique<RunPresenter>(this, new RunView(m_uiForm.runWidget));
   setOutputPlotOptionsPresenter(
       std::make_unique<OutputPlotOptionsPresenter>(m_uiForm.ipoPlotOptions, PlotWidget::Spectra));
 
   connect(m_uiForm.mwInputFile, SIGNAL(filesFound()), this, SLOT(handleFileChange()));
 
-  connect(m_uiForm.pbRun, SIGNAL(clicked()), this, SLOT(runClicked()));
   connect(m_uiForm.pbSave, SIGNAL(clicked()), this, SLOT(saveClicked()));
 
   m_uiForm.lwIons->setSelectionMode(QAbstractItemView::MultiSelection);
 }
 
-void DensityOfStates::setup() {}
-
-/**
- * Validate the form to check the program can be run.
- *
- * @return Whether the form was valid
- */
-bool DensityOfStates::validate() {
-  UserInputValidator uiv;
+void DensityOfStates::handleValidation(IUserInputValidator *validator) const {
+  auto const filename = m_uiForm.mwInputFile->getFirstFilename().toStdString();
+  if (filename.empty()) {
+    validator->addErrorMessage("A data file has not been loaded.");
+    return;
+  }
+  auto const format = filenameToFormat(filename);
+  if (format == InputFormat::Unsupported) {
+    validator->addErrorMessage("The provided file format is unsupported. The supported extensions are 'phonon', "
+                               "'castep', 'castep_bin' and 'yaml'.");
+    return;
+  }
 
-  const auto filename = m_uiForm.mwInputFile->getFirstFilename();
-  InputFormat format = filenameToFormat(filename);
-  QString specType = m_uiForm.cbSpectrumType->currentText();
-  auto items = m_uiForm.lwIons->selectedItems();
+  auto const specType = m_uiForm.cbSpectrumType->currentText();
+  auto const items = m_uiForm.lwIons->selectedItems();
 
   if (specType == "DensityOfStates" && isPdosFile(format) && items.size() < 1)
-    uiv.addErrorMessage("Must select at least one ion for DensityOfStates.");
-
-  // Give error message when there are errors
-  if (!uiv.isAllInputValid())
-    emit showMessageBox(uiv.generateErrorMessage());
-
-  return uiv.isAllInputValid();
+    validator->addErrorMessage("Must select at least one ion for DensityOfStates.");
 }
 
-/**
- * Configures and executes the DensityOfStates algorithm.
- */
-void DensityOfStates::run() {
-  setRunIsRunning(true);
+void DensityOfStates::handleRun() {
+  clearOutputPlotOptionsWorkspaces();
 
   // Get the SimulatedDensityOfStates algorithm
   auto dosAlgo = AlgorithmManager::Instance().create("SimulatedDensityOfStates");
 
-  const auto filename = m_uiForm.mwInputFile->getFirstFilename();
+  const auto filename = m_uiForm.mwInputFile->getFirstFilename().toStdString();
```

Comment:
Just one curiosity and then I am happy to approve. I noticed you changed `Qstring` to a normal string, is there any other reason other than it's preferable to rely on the std library?

Reply:
Another reason I prefer to use `std::string` is because when debugging using Visual Studio, and you want to check the value of a `QString` variable, you have to dig down several layers to find its actual value (when hovering over the variable). Whereas hovering over a std::string will immediately show you its value - so allows quicker/easier debugging

Reply:
Ah brilliant, that makes a lot of sense